File: RetrofitExample/app/src/main/java/com/example/retrofitexample/MainActivity.java
package com.example.retrofitexample;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import com.example.retrofitexample.api.ApiService;
import com.example.retrofitexample.api.SeviceBuilder;
import com.example.retrofitexample.model.User;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MainActivity extends AppCompatActivity {

    private TextView title;
    private TextView price;
    private TextView category;
    private Button button;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        title = findViewById(R.id.title);
        price = findViewById(R.id.price);
        category = findViewById(R.id.category);
        button = findViewById(R.id.button);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                callApi();
            }
        });
    }

    public void callApi() {
        SeviceBuilder.getApiService().getUsers().enqueue(new Callback<List<User>>() {
            @Override
            public void onResponse(Call<List<User>> call, Response<List<User>> response) {
                Toast.makeText(MainActivity.this, "Call api success", Toast.LENGTH_SHORT).show();

                List<User> user = response.body();
                title.setText(user.get(0).getName());
                price.setText(user.get(0).getEmail());
                category.setText(user.get(0).getRole());
            }

            @Override
            public void onFailure(Call<List<User>> call, Throwable t) {
                Toast.makeText(MainActivity.this, t.toString(), Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: Xu_ly_su_kien_button/app/src/main/java/com/example/xu_ly_su_kien_button/MainActivity.kt
package com.example.xu_ly_su_kien_button

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText

class MainActivity : AppCompatActivity() {
    private lateinit var textA : EditText
    private lateinit var textB : EditText
    private lateinit var textResult : EditText

    private lateinit var btnAdd : Button
    private lateinit var btnReset : Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        textA = findViewById(R.id.editTextA)
        textB = findViewById(R.id.editTextB)
        textResult = findViewById(R.id.editTextResult)

        btnAdd = findViewById(R.id.btnAdd)
        btnReset = findViewById(R.id.btnReset)

        btnAdd.setOnClickListener {
            val a: Int = textA.text.toString().toInt()
            val b: Int = textB.text.toString().toInt()
            val result : Int = a + b
            textResult.setText(result.toString())
        }

        btnReset.setOnClickListener {
            textA.setText("")
            textB.setText("")
            textResult.setText("")
        }
    }
}